.github/actions/build-test-image/action.yml

```
@@ -0,0 +1,59 @@
+name: Build Test Image
+description: A composite action that allows building and publishing the test remote runner image
+
+inputs:
+  repository:
+    description: The docker repository for the image
+    default: chainlink-tests
+    required: false
+  tag:
+    description: The tag to use by default and to use for checking image existance
+    default: ${{ github.sha }}
+    required: false
+  other_tags:
+    description: Other tags to push if needed
+    required: false
+  suites:
+    description: The test suites to build into the image
+    default: chaos migration performance reorg smoke soak benchmark
+    required: false
+  QA_AWS_ROLE_TO_ASSUME:
+    description: The AWS role to assume as the CD user, if any. Used in configuring the docker/login-action
+    required: true
+  QA_AWS_REGION:
+    description: The AWS region the ECR repository is located in, should only be needed for public ECR repositories, used in configuring docker/login-action
+    required: true
+  QA_AWS_ACCOUNT_NUMBER:
+    description: The AWS region the ECR repository is located in, should only be needed for public ECR repositories, used in configuring docker/login-action
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Check if image exists
+      id: check-image
+      uses: smartcontractkit/chainlink-github-actions/docker/image-exists@ab595504ae9cf10c60eb8d2c5ce025284e58b210 #v2.1.5
+      with:
+        repository: ${{ inputs.repository }}
+        tag: ${{ inputs.tag }}
+        AWS_REGION: ${{ inputs.QA_AWS_REGION }}
+        AWS_ROLE_TO_ASSUME: ${{ inputs.QA_AWS_ROLE_TO_ASSUME }}
+    - name: Build and Publish Test Runner
+      if: steps.check-image.outputs.exists == 'false'
+      uses: smartcontractkit/chainlink-github-actions/docker/build-push@ab595504ae9cf10c60eb8d2c5ce025284e58b210 #v2.1.5
+      with:
+        tags: |
+          ${{ inputs.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ inputs.QA_AWS_REGION }}.amazonaws.com/${{ inputs.repository }}:${{ inputs.tag }}
+          ${{ inputs.other_tags }}
+        file: ./integration-tests/test.Dockerfile
+        build-args: |
+          BASE_IMAGE=${{ inputs.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ inputs.QA_AWS_REGION }}.amazonaws.com/test-base-image
+          IMAGE_VERSION=v0.3.29
+          SUITES="${{ inputs.suites }}"
+        AWS_REGION: ${{ inputs.QA_AWS_REGION }}
+        AWS_ROLE_TO_ASSUME: ${{ inputs.QA_AWS_ROLE_TO_ASSUME }}
+    - name: Print Image Built
+      shell: sh
+      run: |
+        echo "### ${{ inputs.repository }} image tag for this test run :ship:" >>$GITHUB_STEP_SUMMARY
+        echo "\`${{ inputs.tag }}\`" >>$GITHUB_STEP_SUMMARY
```

Make sure to set this in the step like:

```
      env:
        INPUTS_TAG: ${{ inputs.tag }}
```
and then you can use it like:

```suggestion
        echo "${INPUTS_TAG}" >> $GITHUB_STEP_SUMMARY
```

---

What is the reasoning for this and should I be doing this everywhere?

---

I'd say this is optional but if you wanted to not have to worry about escaping GitHub contexts in bash, it'd be cleaner to pass it in as a bash env var.